pkg/scheduler/objects/application.go

```diff
@@ -973,7 +974,7 @@ func (sa *Application) tryAllocate(headRoom *resources.Resource, allowPreemption
 			// the iterator might not have the node we need as it could be reserved, or we have not added it yet
 			node := getNodeFn(requiredNode)
 			if node == nil {
-				rateLimitedLog.Warn("required node is not found (could be transient)",
+				GetAppLog().Warn("required node is not found (could be transient)",
```

Comment:
Remembering why this was rate-limited in the first place, we really need to use `Info()` here rather than `Warn()`. `Warn()` generates stack traces, which is really not appropriate here, especially when this is a (somewhat) common occurrence. It's nice to know about, but should not spam the logs with stack traces, even when rate limited.

Reply:
That sounds sensible. Thank you for sharing with me this kind of insights, it is very useful.

Reply:
Done.